src/zsys.c

```diff
@@ -150,13 +153,46 @@ s_zsys_vprintf_hint (int hint, const char *format, va_list argptr);
 //  Not threadsafe, so call only from main thread. Safe to call multiple
 //  times. Returns global CZMQ context.
 
+// mutex for pthread_once to run the init function only once in a process
+static pthread_once_t init_all_mutex_var = PTHREAD_ONCE_INIT;
+
+// handler to initialize mutexes one time in multi threaded env
+void zsys_initialize_mutex() {
```

Comment:
all these new functions should be static

Reply:
i will send a new patch after changing these functions to static.